api/service/syncing/downloader/client.go

```diff
@@ -41,10 +41,12 @@ func (client *Client) Close() {
 }
 
 // GetBlockHashes gets block hashes from all the peers by calling grpc request.
-func (client *Client) GetBlockHashes(startHash []byte, size uint32) *pb.DownloaderResponse {
+func (client *Client) GetBlockHashes(startHash []byte, size uint32, ip, port string) *pb.DownloaderResponse {
```

Comment:
why use ip and port in the function here?  It is weird. You'd wrap it up in the 'request' or other object.

Reply:
the request is created inside this function, so I have to pass it as parameter. Let me know if you have better suggestion.